### PR TITLE
(feat) O3-3504 : Implement the tutorial modal and listing tutorials

### PR DESCRIPTION
**Pull request by Vijaykv5**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR implements a `tutorial modal` when clicked on the tutorials from the help menu and lists all the defined tutorials there.

## Screenshots

https://github.com/openmrs/openmrs-esm-user-onboarding/assets/94985341/f53e369d-4c24-4d61-9498-9a96ae41c045

## Related Issue

[O3-3504
](https://openmrs.atlassian.net/browse/O3-3504)

## Other

[Zeplin Design ](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/62c6b9f6d3cea618db0836c8)

### CODE REVIEW

File: src/tutorial/tutorial.tsx [hunk 1]
```diff
@@ -1,11 +1,25 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-
+import Tutorials from './modal.component';
 
 const Tutorial = () => {
   const { t } = useTranslation();
+  const [isModalOpen, setModalOpen] = useState(false);
```
Comment (jayasanka-sack) [inline src/tutorial/tutorial.tsx:7]:
```suggestion
  const [isModalOpen, setIsModalOpen] = useState(false);
```
While it's not incorrect, a more conventional naming pattern is to prefix setter functions with set. matching the name of the state variable.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]

**Comment by Vijaykv5** [comment]
The updates styles looks like this.

https://github.com/openmrs/openmrs-esm-user-onboarding/assets/94985341/2d123a1a-e903-44d9-9650-aa5f848a9ef8

I haven't added the video for now ( as per the [design](https://zpl.io/vM5R4Bj) ), may be we can figure out it later.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Awesome!

> I haven't added the video for now ( as per the [design](https://zpl.io/vM5R4Bj) ), may be we can figure out it later.
Yes, we can think of it later

Some comments on the look and feel, the following should match the design:
1. Main header of the modal 
2. Title of each tutorial - should use heading-02
3. description of each tutorial - color should be gray 70
4. Walkthrough button - increase the gap and font size - body-compact-02/$link-primary

Always try to match the provided designs, you can find the styles and variables on zeplin.

design:
Screenshot 2024-07-04 at 13 06 50

current implementation:
image

**Comment by Vijaykv5** [comment]
Latest Changes :

Screenshot 2024-07-04 at 8 37 27 PM

File: src/tutorial/styles.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      margin-bottom: spacing.$spacing-04;
+      @include type.type-style('heading-02');
+    }
+
+    .tutorialDescription {
+      color: $color-gray-70;
+      @include type.type-style('body-compact-01');
+    }
+
+    .walkthrough {
```
Comment (jayasanka-sack) [inline src/tutorial/styles.scss:29]: This should work here, I guess:

@include type.type-style('body-compact-02');
color: $link-primary;

Comment (Vijaykv5) [reply src/tutorial/styles.scss:29]: is the `color: $link-primary` defined?
I'm getting an error as an Undefined variable.

Comment (jayasanka-sack) [reply src/tutorial/styles.scss:29]: makes sense, even our primary color isn't blue.

Comment (jayasanka-sack) [reply src/tutorial/styles.scss:29]: how about: @include type.type-style('body-compact-02');

Comment (Vijaykv5) [reply src/tutorial/styles.scss:29]: Yeah that works

File: src/tutorial/styles.scss [hunk 3]
```diff
@@ -31,6 +31,7 @@
       margin-top: spacing.$spacing-03;
       color: $color-blue-60-2;
       font-size: 1rem;
+      @include type.type-style('body-compact-02');
```
Comment (jayasanka-sack) [inline src/tutorial/styles.scss:34]: I think font size also included with the `body-compact-02`. Can you please check?

Comment (Vijaykv5) [reply src/tutorial/styles.scss:34]: Yes that's unnecessary. will remove it.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @Vijaykv5 !

**Pull request merged**